### PR TITLE
HDDS-11230. Make RocksDB cache configurable in Ozone Manager.

### PR DESCRIPTION
**Pull request by weimingdiit**
## What changes were proposed in this pull request?
Make RocksDB cache configurable in Ozone Manager.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-11230

## How was this patch tested?
unit tests

### CODE REVIEW

**Comment by weimingdiit** [comment]
There are some test errors, I will update the code

**Comment by errose28** [comment]
Hi @weimingdiit thanks for looking in to this. Maybe it would be better to support RocksDB config files directly instead of needing to expose each config through the Ozone configuration? See the [RocksDB docs](https://github.com/facebook/rocksdb/wiki/RocksDB-Options-File) on this for reference. We could first load the options set by the user in this file, and then override any in Ozone that must be set a certain way, for example creating column families that don't exist.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java [hunk 1]
```diff
@@ -159,6 +159,16 @@ public final class OzoneConfigKeys {
   public static final String
       OZONE_METADATA_STORE_ROCKSDB_CF_WRITE_BUFFER_SIZE_DEFAULT = "128MB";
 
+  public static final String OZONE_METADATA_STORE_ROCKSDB_BLOCK_CACHE_SIZE =
+      "ozone.metastore.rocksdb.block.cache.size";
+  public static final String
+      OZONE_METADATA_STORE_ROCKSDB_BLOCK_CACHE_SIZE_DEFAULT = "256MB";
```
Comment (kerneltime) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java:165]: The default needs to be larger, 1 or 2 GB perhaps?

**Comment by weimingdiit** [comment]
> Hi @weimingdiit thanks for looking in to this. Maybe it would be better to support RocksDB config files directly instead of needing to expose each config through the Ozone configuration? See the [RocksDB docs](https://github.com/facebook/rocksdb/wiki/RocksDB-Options-File) on this for reference. We could first load the options set by the user in this file, and then override any in Ozone that must be set a certain way, for example creating column families that don't exist.

Thank you very much for your comments，this is a good idea. However, for the current version, could we first make some of the more performance-critical parameters configurable, such as the ones I mentioned? I will continue to follow up on this issue and create a new issue to directly support RocksDB configuration files, so we don't have to rely on Ozone configuration. What do you think?

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java (hunk 1 above)
Comment (weimingdiit) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java:165]: Indeed, this parameter should be larger. For larger tables, 1 to 2 GB is a suitable default value. I will update this parameter.  Hi,  @errose28  What do you think about it?

**Comment by weimingdiit** [comment]
@jojochuang  Could you help review this PR? This PR is related to some caching in RocksDB. My current thought is to keep the status quo unchanged in this PR and then create a new issue to make all RocksDB parameters configurable. I'd like to hear your thoughts on this—what do you think?

**Comment by weimingdiit** [comment]
> > Hi @weimingdiit thanks for looking in to this. Maybe it would be better to support RocksDB config files directly instead of needing to expose each config through the Ozone configuration? See the [RocksDB docs](https://github.com/facebook/rocksdb/wiki/RocksDB-Options-File) on this for reference. We could first load the options set by the user in this file, and then override any in Ozone that must be set a certain way, for example creating column families that don't exist.
> 
> @errose28 Thank you very much for your comments，this is a good idea. However, for the current version, could we first make some of the more performance-critical parameters configurable, such as the ones I mentioned?

I thought about it, maybe it would be more convenient for us to manage it through ozone configuration, so that our configuration will have only one file, which is easier to manage.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java (hunk 1 above)
Comment (errose28) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java:165]: That seems fine to me. I don't have data on the size of serialized proto values but that could help approximate the number of keys the cache would hold.

**Comment by errose28** [comment]
I'm ok with adding this field in the Ozone configuration to start if there are no immediate plans to support the full RocksDB INI file. We can file that as a follow up Jira. In the future I do think we should support the INI file instead of propagating every RocksDB config we might need into the Ozone configuration. If we start getting into serious RocksDB tuning then there will be too many options in RocksDB for that approach to be sustainable long term.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java (hunk 1 above)
Comment (weimingdiit) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java]: @kerneltime  The default size has been updated to 1GB, thanks for your comment

**Comment by weimingdiit** [comment]
> I'm ok with adding this field in the Ozone configuration to start if there are no immediate plans to support the full RocksDB INI file. We can file that as a follow up Jira. In the future I do think we should support the INI file instead of propagating every RocksDB config we might need into the Ozone configuration. If we start getting into serious RocksDB tuning then there will be too many options in RocksDB for that approach to be sustainable long term.

**Comment by weimingdiit** [comment]
> I'm ok with adding this field in the Ozone configuration to start if there are no immediate plans to support the full RocksDB INI file. We can file that as a follow up Jira. In the future I do think we should support the INI file instead of propagating every RocksDB config we might need into the Ozone configuration. If we start getting into serious RocksDB tuning then there will be too many options in RocksDB for that approach to be sustainable long term.

@errose28  I will close this PR, and then make all RocksDB parameters configurable in https://github.com/apache/ozone/pull/7150

**Pull request closed without merging**